app/Features/QuerySpecificAnimeSeasonHandler.php

```diff
@@ -27,7 +28,7 @@ protected function getSeasonItems($request, ?AnimeTypeEnum $type): Builder
 
         [$from, $to] = $this->getSeasonRange($request->year, $request->season);
 
-        return $this->repository->getAiredBetween($from, $to, $type);
+        return $this->repository->getAiredBetween($from, $to, $type, $request->season, $request->year);
```

Comment:
The `getAiredBetween` should have some special logic instead, where it uses it's arguments on the database fields correctly.

Reply:
I've modified getAiredBetween to take `premiered` as an argument and handle the logic based on that.